src/native/libs/System.IO.Compression.Native/extra_libs.cmake

```diff
@@ -1,8 +1,8 @@
 macro(append_extra_compression_libs NativeLibsExtra)
   set(ZLIB_LIBRARIES "")
   # TODO: remove the mono-style HOST_ variable checks once Mono is using eng/native/configureplatform.cmake to define the CLR_CMAKE_TARGET_ defines
-  if (CLR_CMAKE_TARGET_BROWSER OR HOST_BROWSER OR CLR_CMAKE_TARGET_WASI OR HOST_WASI)
-      # nothing special to link
+  if (CLR_CMAKE_HOST_ARCH_WASM OR CLR_CMAKE_TARGET_BROWSER OR HOST_BROWSER OR CLR_CMAKE_TARGET_WASI OR HOST_WASI)
+      list(APPEND ZLIB_LIBRARIES z)
```

Comment:
I could not use `find_package( ZLIB REQUIRED)` before line 5 because I would get this error:

```
  CMake Error at /usr/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
    Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
  Call Stack (most recent call first):
    /usr/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
    /usr/share/cmake-3.28/Modules/FindZLIB.cmake:199 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    /home/carlos/repos/runtime/src/native/libs/System.IO.Compression.Native/extra_libs.cmake:5 (find_package)
    CMakeLists.txt:517 (append_extra_compression_libs)
```

In my docker container that imitates what we do in the CI, I can confirm that libz is available under the crossrootfs folder:

```
# ls -l /crossrootfs/x64/lib/x86_64-linux-gnu/*libz*
lrwxrwxrwx 1 root root     13 Jan 21  2020 /crossrootfs/x64/lib/x86_64-linux-gnu/libz.so.1 -> libz.so.1.2.8
-rw-r--r-- 1 root root 104864 Jan 21  2020 /crossrootfs/x64/lib/x86_64-linux-gnu/libz.so.1.2.8
```

So I figured maybe this is a similar case to the android branch, where find_package would conflict, so I tried by just adding the `list` call too. Am I missing something?